test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -420,4 +429,24 @@ public void testUpdates() throws IOException {
         assertParseMaximalWarnings();
     }
 
+    public final void testTextSearchInfoConsistency() throws IOException {
```

Comment:
Can we put this on FieldTypeTestCase rather than MapperTestCase? Given that we're explicitly checking things in MappedFieldType rather than FieldMapper I think it would work better there.

Reply:
that makes sense, but it would require replicating quite some logic in both test base classes. I am reusing some of the existing methods here that are missing in FieldTypeTestCase. What do you think?

Reply:
cc @nik9000 - I think it's still worth trying to keep things separate?

Another possibility is to actually make this a constraint by moving the termQuery() etc factory functions onto TextSearchInfo.

Reply:
I think it'd make sense to merge `FieldTypeTestCase` into `MapperTestCase`.  It might also make sense to move the query builder methods but one thing at a time!

Reply:
I would prefer addressing where the new test should be at a later stage. I am not sure it is worth duplicating all of the needed testing logic to `FieldTypeTestCase`, and the other proposed alternatives look like potential follow-ups too.